src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs

```diff
@@ -194,7 +194,7 @@ public void TlsHandshakeStop(BaseConnectionContext connectionContext, TlsConnect
             // TODO: Write this without a string allocation using WriteEventData
             var applicationProtocol = feature == null ? string.Empty : Encoding.UTF8.GetString(feature.ApplicationProtocol.Span);
             var sslProtocols = feature?.Protocol.ToString() ?? string.Empty;
-            var hostName = feature?.HostName ?? string.Empty;
+            var hostName = feature?.HostName;
```

Comment:
Revert. You still need the fallback in case the feature is null.
```suggestion
            var hostName = feature?.HostName ?? string.Empty;
```

Reply:
right, reverted. thanks